### PR TITLE
AL-552: Tests and step code needed to change to accommodate for new parameter in `ramp_fit`.

### PR DESCRIPTION
**Pull request by kmacdonald-stsci**
Resolves AL-552

**Description**

To make ramp fitting code pipeline agnostic DQ flags are now passed as a parameter to `ramp_fit`.  The new parameter to `ramp_fit` required changes to the tests and step code.  All JWST ramp fit unit tests and regression tests pass with these changes.

Checklist
- [x] Tests
- [ ] Documentation
- [x] Change log
- [x] Milestone
- [x] Label(s)

### CODE REVIEW

File: jwst/ramp_fitting/ramp_fit_step.py
Comment (jdavies-st) [inline jwst/ramp_fitting/ramp_fit_step.py:241]: I think originally some of these DQ flags were from `jwst.datamodels.dqflags.group` dict.

https://github.com/spacetelescope/jwst/blob/f8d9122b78dd4c4f34df5ca0199e0857116ed0a3/jwst/ramp_fitting/ramp_fit.py#L33-L34

And some from the pixel dict:

https://github.com/spacetelescope/jwst/blob/f8d9122b78dd4c4f34df5ca0199e0857116ed0a3/jwst/ramp_fitting/ramp_fit.py#L1547

Though it seems the group and pixel dq flags are identical.  (above code snippets from release 1.1.0)

Is there a reason to redefine them here in the calling code instead of just shuffling the full dictionary on to `ramp_fit` and letting it use them as it used them before?

File: jwst/ramp_fitting/tests/test_ramp_fit.py
Comment (jdavies-st) [inline jwst/ramp_fitting/tests/test_ramp_fit.py:16]: Again, no need for this.  Just use the existing dicts in `jwst.datamodels.dqflags.pixel` and `jwst.datamodels.dqflags.group`.

File: jwst/ramp_fitting/tests/test_ramp_fit_cases.py
Comment (jdavies-st) [inline jwst/ramp_fitting/tests/test_ramp_fit_cases.py:28]: Same here.

File: jwst/ramp_fitting/ramp_fit_step.py
Comment (kmacdonald-stsci) [reply jwst/ramp_fitting/ramp_fit_step.py:241]: The `group` flags are defined from the `pixel` dictionary.  I simply created these test DQ flag dictionaries as a demonstration that the values could be defined to be anything, so long as they are consistent, as well as to show the required key words to pass to `ramp_fit`.

Comment (kmacdonald-stsci) [reply jwst/ramp_fitting/ramp_fit_step.py:241]: I changed the call to `ramp_fit` to simply pass `dqflags.pixel`, rather than create a separate dictionary to pass.

File: jwst/ramp_fitting/tests/test_ramp_fit.py
Comment (kmacdonald-stsci) [reply jwst/ramp_fitting/tests/test_ramp_fit.py:16]: I changed the code to `test_dq_flags = dqflags.pixel`.

File: jwst/ramp_fitting/tests/test_ramp_fit_cases.py
Comment (kmacdonald-stsci) [reply jwst/ramp_fitting/tests/test_ramp_fit_cases.py:28]: I changed the code to `test_dq_flags = dqflags.pixel`.

**Review by jdavies-st — APPROVED** [review]
This looks good.

This PR assumes that `dqflags.group` dict is a subset of `dqflags.pixel` and that those mnemonics are interchangeable.  Which they _currently_ are.  And that was the intention, but that is not actually formalized.

So I think this is OK.

Approving, though merging will have to wait until all the changes in `stcal` are merged and released.

Will file a separate issue/PR for formalize `dqflags.pixel` being a superset of `dqflags.group`.

**Comment by jdavies-st** [comment]
FWIW, `dqflags.group` being a subset of `dqflags.pixel` was formalized in #4726

File: jwst/ramp_fitting/ramp_fit_step.py
Comment (hbushouse) [inline jwst/ramp_fitting/ramp_fit_step.py:169]: I think that calling .update will include the copying of meta.filename from the input to the output model, so line 168 may not even be necessary. Also, do we really want to set the input and output file names to be the same? Shouldn't the opt_model get saved to a file name that at least has a different file type suffix? Or is that handled later?

**Review by hbushouse — COMMENTED** [review]
Needs a change log entry (in CHANGES.rst)

File: jwst/ramp_fitting/ramp_fit_step.py
Comment (jdavies-st) [reply jwst/ramp_fitting/ramp_fit_step.py:169]: That's done later in the Step code:

https://github.com/spacetelescope/jwst/blob/8a8f148d4ab6310282ac76162890d44b81122840/jwst/ramp_fitting/ramp_fit_step.py#L240-L243

**Comment by jdavies-st** [comment]
Unit and regression tests pass pointing to `stcal 0.2.2`

https://plwishmaster.stsci.edu:8081/job/RT/job/jdavies-dev/262/

Merging.

**Pull request merged**
